vtex-apps/login
# Auto-redirect to OAuth when there's only one provider

#### What is the purpose of this pull request?
This PR adds an auto-redirect to an OAuth provider when it is the only login option available for the store. It does it by replacing the `<LoginOptions />` component and informing the user he/she is being redirected to an external site.

>Note: The best use of this functionality is achieved when the store theme is also configured to hide the login form in the Login Page by setting the prop `isInitialScreenOptionOnly` to true.

This PR also adds a loading component to the login options.

#### What problem is this solving?
Some stores want to have a cross-site login using our built-in OAuth solution. In order to improve the user experience, they don't the user to have to open the login modal or page and then click on the OAuth provider option. They would like to be automatic.

#### How should this be manually tested?
To test the app with the redirect, go to [parceiro01's home](https://bezerra--parceiro01.myvtex.com/) or [parceiro01's login page](https://bezerra--parceiro01.myvtex.com/login). This account has only facebook enabled and should redirect you there.
To test the app without the redirect, goto [storecomponents's home](https://bezerra--storecomponents.myvtex.com/) or [storecomponents's login page](https://bezerra--storecomponents.myvtex.com/login). This account has more than one OAuth provider, and *should not* redirect you.


#### Screenshots or example usage
##### Login Modal (Home page) Before 
![oauth-redirect-home-before](https://user-images.githubusercontent.com/8474847/57702430-71bb7b80-7634-11e9-9e3e-91751c0cf820.gif)
##### Login Modal (Home page) After
![oauth-redirect-home-after](https://user-images.githubusercontent.com/8474847/57702484-9283d100-7634-11e9-9e43-cfb9c03d020d.gif)

##### Login Page Before 
![oauth-redirect-login-2-before](https://user-images.githubusercontent.com/8474847/57702594-c828ba00-7634-11e9-8aa4-672009fc9261.gif)
##### Login Page After 
![oauth-redirect-login-2-after](https://user-images.githubusercontent.com/8474847/57702613-d2e34f00-7634-11e9-89ad-43b4fb0446c0.gif)


#### Types of changes
